src/Utility/Composer.php

```diff
@@ -67,25 +86,27 @@ public static function createProject(
             ));
         }
 
-        // Update with the given repository
-        $updateOptions = self::getUpdateOptions($preferDist, $ignorePlatform);
-        $runner->runCommand(array_merge([
-            'composer',
-            'update',
-            '--working-dir',
-            $directory,
-        ], $updateOptions), "Could not update project");
-
-        // Revert all custom config
-        foreach ($customConfig as $option => $arguments) {
-            $runner->runCommand([
+        try {
+            // Update with the given repository
+            $updateOptions = self::getUpdateOptions($preferDist, $ignorePlatform);
+            $runner->runCommand(array_merge([
                 'composer',
-                'config',
-                '--unset',
-                $option,
+                'update',
                 '--working-dir',
                 $directory,
-            ]);
+            ], $updateOptions), "Could not update project");
+        } finally {
+            // Revert all custom config
+            foreach ($customConfig as $option => $arguments) {
+                $runner->runCommand([
+                    'composer',
+                    'config',
+                    '--unset',
+                    $option,
+                    '--working-dir',
+                    $directory,
+                ]);
+            }
         }
```

Comment:
This split is technically an API break, but I don't think we mind too much since cow is mostly an internal tool.